docs/general/builtins/scalar-functions.rst

```diff
@@ -4243,6 +4243,64 @@ Special functions
 =================
 
 
+.. _scalar_knn_match:
+
+
+``knn_match(float_vector, float_vector, int)``
+----------------------------------------------
+
+The ``knn_match`` function uses a k-nearest
+neighbour (kNN) search algorithm to find vectors that are similar
+to a query vector.
+
+The first argument is the column to search.
+The second argument is the query vector.
+The third argument is the number of nearest neighbours to search and return.
+
+``knn_match(search_vector, target, k)``
+
+This function must be used within a ``WHERE`` clause targeting a table to use it
+as a predicate that searches the whole dataset of a table.
+If used *outside* of a ``WHERE`` clause, or in a ``WHERE`` clause targeting a
+virtual table instead of a physical table, the search algorithm will only
+consider the current row, not the whole dataset.
+
+Similar to the :ref:`MATCH predicate <predicates_match>`, this function affects
+the :ref:`_score <sql_administration_system_column_score>` value.
```

Comment:
Not sure what kind of distance Lucene uses by default, I found https://solr.apache.org/guide/8_8/machine-learning.html#k-nearest-neighbor-knn so I think it's Euclidean but it could compute Manhattan and some others.

Since we don't provide distance type as argument we probably need to check and mention  that `_score` holds  (whatever_is_default_most_likely_Euclidean) distance between vectors

We can say it's obvious and users can check it themselves by computing Euclidean on their own and checking _score output but would be nice to have a clear statement in the docs.

Reply:
My plan was to do another follow up and see if we can make this configurable via a type parameter.